src/dev/code_coverage/ingest_coverage/transforms.js

```diff
@@ -0,0 +1,165 @@
+
+export const statsAndstaticSiteUrl = (...xs) => {
+  const [staticSiteUrl] = xs[0][1];
+  const [stats] = xs[0];
+  return {
+    staticSiteUrl,
+    ...stats,
+  };
+};
+
+export const addJsonSummaryPath = jsonSummaryPath => obj => ({
+  jsonSummaryPath: trimLeftFrom('target', jsonSummaryPath),
+  ...obj,
+});
+
+export const truncate = text => obj => {
+  const { staticSiteUrl } = obj;
+  if (staticSiteUrl.includes(text)) obj.staticSiteUrl = trimLeftFrom(text, staticSiteUrl);
+  return obj;
+};
+
+export const addTimeStamp = ts => obj => ({
+  ...obj,
+  '@timestamp': ts,
+});
+
+const splitOnRoot = root => x => x.split(root)[1];
+const setTotal = x => obj => (obj.isTotal = x);
+const mutateTrue = setTotal(true);
+const mutateFalse = setTotal(false);
+
+const root = urlBase => ts => testRunnerType =>
+  `${urlBase}/${ts}/${testRunnerType.toLowerCase()}-combined`;
+
+const prokForTotalsIndex = mutateTrue => urlRoot => obj =>
+  right(obj)
+    .map(mutateTrue)
+    .map(always(`${urlRoot}/index.html`))
+    .fold(noop, id);
+
+const prokForCoverageIndex = root => mutateFalse => urlRoot => obj => siteUrl =>
+  right(siteUrl)
+    .map(x => {
+      mutateFalse(obj);
+      return x;
+    })
+    .map(x => {
+      return splitOnRoot(root)(x);
+    })
+    .map(x => `${urlRoot}${x}.html`)
+    .fold(noop, id);
+
+export const staticSite = urlBase => obj => {
+  const { staticSiteUrl, testRunnerType, COVERAGE_INGESTION_KIBANA_ROOT } = obj;
+  const ts = obj['@timestamp'];
+  const urlRoot = root(urlBase)(ts)(testRunnerType);
+  const prokTotal = prokForTotalsIndex(mutateTrue)(urlRoot);
+  const prokCoverage = prokForCoverageIndex(COVERAGE_INGESTION_KIBANA_ROOT)(mutateFalse)(urlRoot)(
+    obj
+  );
+  const prokForBoth = always(maybeTotal(staticSiteUrl).fold(always(prokTotal(obj)), prokCoverage));
+
+  return { ...obj, staticSiteUrl: prokForBoth() };
+};
+
+export const coveredFilePath = obj => {
+  const { staticSiteUrl, COVERAGE_INGESTION_KIBANA_ROOT } = obj;
+
+  const withoutCoveredFilePath = always(obj);
+  const leadingSlashRe = /^\//;
+  const maybeDropLeadingSlash = x => (leadingSlashRe.test(x) ? right(x) : left(x));
+  const dropLeadingSlash = x => x.replace(leadingSlashRe, '');
+  const dropRoot = root => x =>
+    maybeDropLeadingSlash(splitOnRoot(root)(x)).fold(id, dropLeadingSlash);
+  return maybeTotal(staticSiteUrl)
+    .map(dropRoot(COVERAGE_INGESTION_KIBANA_ROOT))
+    .fold(withoutCoveredFilePath, coveredFilePath => ({ ...obj, coveredFilePath }));
+};
+
+export const ciRunUrl = obj => {
+  const ciRunUrl = process.env.CI_RUN_URL || 'CI RUN URL NOT PROVIDED';
+
+  return {
+    ...obj,
+    ciRunUrl,
+  };
+};
+
+const dropAfter = size => x => {
+  return x.slice(0, size);
+};
+
+const truncateCommitMsg = x => {
+  const size = 50;
+  const dropAfterSize = dropAfter(size);
+  const trunkd = x.length > 50 ? `${dropAfterSize(x)}...` : x;
```

Comment:
Length will never be greater than 50 since you're trimming it to 50 the line prior. Also update to use the `size` variable.

Reply:
Sweet!

Reply:
Actually, the line before it is lazy, not executed, until  the `trunkd` variable is assigned.

Reply:
Gotcha